app/decorators/physical_switch_decorator.rb

```diff
@@ -0,0 +1,42 @@
+class PhysicalSwitchDecorator < MiqDecorator
+  def self.fonticon
+    'ff ff-network-switch'
+  end
+
+  def quadicon
+    {
+      :top_left     => {
+        :text    => (hardware.physical_ports ? hardware.physical_ports.size : 0),
```

Comment:
What about calling `try` instead? It might return a `nil`, but that can be converted to zero using `to_i`:
```ruby
hardware.physical_ports.try(:size).to_i
```

Reply:
Done. I will update in the next minutes.